File: src/actionlib_basics/src/fibonacci_client.cpp
#include <ros/ros.h>
#include <actionlib/client/simple_action_client.h>
#include <actionlib/client/terminal_state.h>
#include <actionlib_basics/FibonacciAction.h>

int main(int argc, char **argv) {
    ros::init(argc, argv, "fibonacci_client");

    auto action_client = actionlib::SimpleActionClient<actionlib_basics::FibonacciAction>("fibonacci", true);

    ROS_INFO("Waiting for action server to start...");
    action_client.waitForServer();

    ROS_INFO("Action server started, sending goal.");
    auto goal = actionlib_basics::FibonacciGoal();
    goal.order = 20;
    action_client.sendGoal(goal);

    // wait for the action to return
    bool finished_before_timeout = action_client.waitForResult(ros::Duration(40.0));
    if (finished_before_timeout) {
        auto state = action_client.getState();
        ROS_INFO("Action finished: %s", state.toString().c_str());

        auto result = action_client.getResult();
        auto sequence = result->sequence;
        int i = 0;
        for (const auto& element : sequence) {
            ROS_INFO("fibonacci(%i) = %i", i, element);
            i += 1;
        }
    } else {
        ROS_INFO("Action did not finish before the time out.");
    }
}

File: src/actionlib_basics/scripts/fibonacci_client.py
#!/usr/bin/env python3

import rospy
import actionlib
import actionlib_basics.msg


def main():
    rospy.init_node("fibonacci_client")
    action_client = actionlib.SimpleActionClient("fibonacci", actionlib_basics.msg.FibonacciAction)

    rospy.loginfo("Waiting for action server to start...")
    action_client.wait_for_server()

    rospy.loginfo("Action server started, sending goal.")
    goal = actionlib_basics.msg.FibonacciGoal()
    goal.order = 20
    action_client.send_goal(goal)

    finished_before_timeout = action_client.wait_for_result(rospy.Duration(40.0))
    if finished_before_timeout:
        state = action_client.get_state()
        rospy.loginfo(f"Action finished: {state}")

        result: actionlib_basics.msg.FibonacciResult = action_client.get_result()
        sequence = result.sequence
        i = 0
        for element in sequence:
            rospy.loginfo(f"fibonacci({i}) = {element}")
            i += 1
    else:
        rospy.logwarn("Action did not finish before the time out.")


if __name__ == "__main__":
    main()
